.github/workflows/codespell.yml

```diff
@@ -0,0 +1,22 @@
+name: Codespell
+on:
+  pull_request:
+permissions:
+  contents: read
+
+jobs:
+  codespell:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Harden Runner
+        uses: step-security/harden-runner@55d479fb1c5bcad5a4f9099a5d9f37c8857b2845 # v2.4.1
+        with:
+          egress-policy: audit
+
+      - uses: actions/checkout@c85c95e3d7251135ab7dc9ce3241c5835cc595a9 # v3.5.3
+      - uses: codespell-project/actions-codespell@94259cd8be02ad2903ba34a22d9c13de21a74461 # master
```

Comment:
> So the latest codespell finds the typos that this PR fixes?

It looks like the latest codespell, `v2.2.6` does find the typos this PR aims to fix. 

> We also run codespell in the make verify target, so we'd run it twice in CI.

You are right. It is better if I update the codespell version in the Makefile instead of adding the `codespell.yml` github action.